kedro-org/kedro
# Add coverage report to help debugging

> **NOTE:** Kedro datasets are moving from `kedro.extras.datasets` to a separate `kedro-datasets` package in
> [`kedro-plugins` repository](https://github.com/kedro-org/kedro-plugins). Any changes to the dataset implementations
> should be done by opening a pull request in that repository.
## Description

We use `coverage` to check which part of the source code is covered. It is useful to answer this question sometimes.
> Which test run this lines?

To test this
1. Run some tests
2. `make show-coverage`
3. Open the `index.html` in a browser or Right-click Open with live server (On GitPod)

You should see a index page. If you click into a specific module, it will shows you which tests execute that line of code.
![image](https://github.com/kedro-org/kedro/assets/18221871/7f7d9968-1992-4dec-ae9a-c28aa98150df)

![image](https://github.com/kedro-org/kedro/assets/18221871/7985289b-da49-48e3-a739-00137fc0810f)



## Development notes

- Add new flag `--cov-context`
- Add a helper function `make show-coverage


## Developer Certificate of Origin
We need all contributions to comply with the [Developer Certificate of Origin (DCO)](https://developercertificate.org/). All commits must be signed off by including a `Signed-off-by` line in the commit message. [See our wiki for guidance](https://github.com/kedro-org/kedro/wiki/Guidelines-for-contributing-developers/).

If your PR is blocked due to unsigned commits, then you must follow the instructions under "Rebase the branch" on the GitHub Checks page for your PR. This will retroactively add the sign-off to all unsigned commits and allow the DCO check to pass.

## Checklist
